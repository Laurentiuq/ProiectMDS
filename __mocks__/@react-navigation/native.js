const navigation = jest.mock('@react-navigation/native', () => {
    return {
        useNavigation: () => {
            return {
                navigate: jest.fn(),
                setOptions: jest.fn(),
            };
        },
        push: () => jest.fn(),

    };
});

const useNavigation = () => {
    return {
        navigate: jest.fn(),
        setOptions: jest.fn(),
    };
}

const push = () => jest.fn();
export { useNavigation, push };
export default navigation;