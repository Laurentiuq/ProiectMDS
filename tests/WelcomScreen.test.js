import React from 'react';
import { render, fireEvent, act, waitFor } from '@testing-library/react-native';

import WelcomeScreen from '../screens/WelcomScreen';
import { useNavigation } from '@react-navigation/native';

jest.mock('@react-navigation/native', () => ({
    ...jest.requireActual('@react-navigation/native'),
    useNavigation: () => ({
        navigate: jest.fn(),
        replace: jest.fn(),
    }),
}));

describe('WelcomeScreen', () => {
    it('renders correctly', () => {
        const { toJSON } = render(<WelcomeScreen />);
        expect(toJSON()).toMatchSnapshot();
    });

    it('shows ActivityIndicator', () => {
        const { getByTestId } = render(<WelcomeScreen />);
        expect(getByTestId('activity-indicator')).toBeTruthy();
    });

    it('navigates to LoginOrSignupScreen after timeout', () => {
        const mockNavigation = { navigate: jest.fn(), replace: jest.fn() };
        jest.spyOn(require('@react-navigation/native'), 'useNavigation').mockReturnValue(mockNavigation);
        jest.spyOn(global, 'setTimeout').mockImplementation((cb) => cb());
        render(<WelcomeScreen />);
        expect(mockNavigation.replace).toHaveBeenCalledWith('LoginOrSignupScreen');
    });

    afterEach(() => {
        jest.clearAllMocks();
    });
});
