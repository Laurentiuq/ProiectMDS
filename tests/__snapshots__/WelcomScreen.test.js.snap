// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeScreen renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "height": "100%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <Image
    resizeMode="cover"
    source={
      {
        "testUri": "../../../assets/mainBackground.png",
      }
    }
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "height": "100%",
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <Image
    source={
      {
        "testUri": "../../../assets/AppLogo.png",
      }
    }
    style={
      {
        "left": -10,
      }
    }
  />
  <Text>
    <Text>
      We quiz therfore we are
    </Text>
  </Text>
  <View
    height={16}
  />
  <ActivityIndicator
    color="#FFFFFF"
    size="large"
    testID="activity-indicator"
  />
</View>
`;
